iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber.hpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_HPP
+
+#include "iceoryx_hoofs/internal/posix_wrapper/mutex.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
+#include "iceoryx_posh/popo/subscriber.hpp"
+
+#include "owl/types.hpp"
+
+#include <limits>
+#include <memory>
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+class EventSubscriber
+{
+  public:
+    using SampleType = T;
+
+    static constexpr uint64_t QUEUE_CAPACITY{10U};
+    static constexpr uint64_t HISTORY_REQUEST{1U};
+    static constexpr bool NOT_OFFERED_ON_CREATE{false};
+
+    EventSubscriber(const core::String& service, const core::String& instance, const core::String& event) noexcept
+        : m_subscriber({service, instance, event},
+                       {QUEUE_CAPACITY, HISTORY_REQUEST, iox::NodeName_t(), NOT_OFFERED_ON_CREATE})
+    {
+    }
+
+    void Subscribe(std::size_t) noexcept
+    {
+        /// @todo #1332 maxSampleCount shall not be ignored, implement getOptions() for user ports?
```

Comment:
@elBoberido 

2) Any objections to adding a `getOptions()` call? Not in this PR.

Reply:
@mossmaurice I'm not quite sure if this solves the problem. Maybe you need to have an `iox::cxx::optional<Subscriber>`, then you could pass the size to the options and create a subscriber which obeys the request.

Reply:
@elBoberido My idea was to give the user a hint if `m_subscriber.getOptions().queueCapacity` would be smaller than the value passed as a parameter to `Subscribe`. However, ate initalization is a much better idea. I'll add it as a comment for a binding implementer.

Reply:
@mossmaurice well, it depends whats the contract of the parameter. For an example it would be enough to print a message but it would also nice to show the user how this actually would be implemented since we cannot specify the queue size on subscribe.

Reply:
Wrapped the `m_subscriber` in a `cxx::optional`.